### PR TITLE
Document built-in exceptions

### LINKED ISSUE #872: Where can I find all the built-in exception documentation, like Not_found?
Hi, everyone

I couldn't find documentation for all the built-in exceptions in the documents and API. Sometimes, I want to use them, but I don't know which exception I should use, what parameters this exception has, and how I should use them.

I hope you can tell me where there are explanations for these exceptions. If there is no documentation, I also hope you can tell me where in the code I can look for them. 

Is it the best practice of exception to define custom exception types and use them instead of build-in exceptions ?

### PR DESCRIPTION
**Pull request by aspeddro**
Close #872

### CODE REVIEW

File: pages/docs/manual/latest/exception.mdx [hunk 1]
```diff
@@ -6,9 +6,36 @@ canonical: "/docs/manual/latest/exception"
 
 # Exception
 
-Exceptions are just a special kind of variant, thrown in **exceptional** cases (don't abuse them!).
+Exceptions are just a special kind of variant, thrown in **exceptional** cases (don't abuse them!). Consider using the `option` or `result` type for recoverable errors.
```
Comment (fhammerschmidt) [inline pages/docs/manual/latest/exception.mdx:9]: Link to the corresponding docs page(s) for Option and Result please.

File: pages/docs/manual/latest/exception.mdx [hunk 2]
```diff
@@ -109,25 +134,234 @@ if (exit === 1) {
 
 </CodeTab>
 
-You can also make your own exceptions like you'd make a variant (exceptions need to be capitalized too).
+### `Invalid_argument`
+
+Used to check if argument is valid. This exception take a string
```
Comment (fhammerschmidt) [inline pages/docs/manual/latest/exception.mdx:139]: takes a string.

**Review by fhammerschmidt — CHANGES_REQUESTED** [review]

File: pages/docs/manual/latest/exception.mdx (hunk 1 above)
Comment (aspeddro) [reply pages/docs/manual/latest/exception.mdx:9]: we don't have documented the `result` builtin type

Comment (aspeddro) [reply pages/docs/manual/latest/exception.mdx:9]: maybe add `Result` page after `Null, Undefined and Option`

Comment (aspeddro) [reply pages/docs/manual/latest/exception.mdx]: Added link to option https://github.com/rescript-association/rescript-lang.org/pull/880/commits/8544ec9f27672d1c67d210d3bebff5fe2cd7842e

File: pages/docs/manual/latest/exception.mdx (hunk 2 above)
Comment (aspeddro) [reply pages/docs/manual/latest/exception.mdx]: Fixed https://github.com/rescript-association/rescript-lang.org/pull/880/commits/8544ec9f27672d1c67d210d3bebff5fe2cd7842e

File: pages/docs/manual/latest/exception.mdx (hunk 1 above)
Comment (fhammerschmidt) [reply pages/docs/manual/latest/exception.mdx]: Thanks. Yes we need to document Result somewhere, but we can fix the link afterwards.
Or just link to the Core Result Api page for now?

Comment (aspeddro) [reply pages/docs/manual/latest/exception.mdx]: 1df7faf3

File: pages/docs/manual/latest/exception.mdx [hunk 3]
```diff
@@ -109,25 +134,234 @@ if (exit === 1) {
 
 </CodeTab>
 
-You can also make your own exceptions like you'd make a variant (exceptions need to be capitalized too).
+### `Invalid_argument`
+
+Used to check if argument is valid. This exception takes a string
```
Comment (fhammerschmidt) [inline pages/docs/manual/latest/exception.mdx:139]: don't forget the `.`

File: pages/docs/manual/latest/exception.mdx [hunk 4]
```diff
@@ -109,25 +134,234 @@ if (exit === 1) {
...
-var InputClosed = Caml_exceptions.create("MyFile.InputClosed");
+### `Failure`
 
-throw {
-  RE_EXN_ID: InputClosed,
-  _1: "The stream has closed!",
-  Error: new Error()
-};
+Exception raised to signal that the given arguments do not make sense. This
+exception take a string as argument.
```
Comment (fhammerschmidt) [inline pages/docs/manual/latest/exception.mdx:273]: takes a string as an argument

**Review by fhammerschmidt — CHANGES_REQUESTED** [review]

File: pages/docs/manual/latest/exception.mdx (hunk 3 above)
Comment (aspeddro) [reply pages/docs/manual/latest/exception.mdx]: https://github.com/rescript-association/rescript-lang.org/pull/880/commits/2680345f43f67314928f06eea32ae5261ab0cf72

File: pages/docs/manual/latest/exception.mdx (hunk 4 above)
Comment (aspeddro) [reply pages/docs/manual/latest/exception.mdx]: https://github.com/rescript-association/rescript-lang.org/pull/880/commits/2680345f43f67314928f06eea32ae5261ab0cf72

**Review by fhammerschmidt — APPROVED** [review]

**Pull request merged**